cmd/checkpoint/main.go

```diff
@@ -19,234 +21,499 @@ import (
 	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/release_1_5"
 	"k8s.io/kubernetes/pkg/client/unversioned/clientcmd"
 	clientcmdapi "k8s.io/kubernetes/pkg/client/unversioned/clientcmd/api"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/runtime"
 )
 
 const (
+	nodeNameEnv = "NODE_NAME"
+
 	kubeletAPIPodsURL = "http://127.0.0.1:10255/pods"
-	ignorePath        = "/srv/kubernetes/manifests"
-	activePath        = "/etc/kubernetes/manifests"
-	manifestFilename  = "apiserver.json"
-	kubeconfigPath    = "/etc/kubernetes/kubeconfig"
-	secretsPath       = "/etc/kubernetes/checkpoint-secrets"
-)
 
-var (
-	tempAPIServer      = []byte("temp-apiserver")
-	kubeAPIServer      = []byte("kube-apiserver")
-	activeManifest     = filepath.Join(activePath, manifestFilename)
-	checkpointManifest = filepath.Join(ignorePath, manifestFilename)
-	secureAPIAddr      = fmt.Sprintf("https://%s:%s", os.Getenv("KUBERNETES_SERVICE_HOST"), os.Getenv("KUBERNETES_SERVICE_PORT_HTTPS"))
+	activeCheckpointPath   = "/etc/kubernetes/manifests"
```

Comment:
This must map to the `--config` or `--pod-manifest-path` for the static pods to be started. The path by convention (and our installation tools) is set to `/etc/kubernetes/manifests`